sql/plan/recursive_cte.go

```diff
@@ -100,51 +99,83 @@ func (r *RecursiveCte) Schema() sql.Schema {
 
 // RowIter implements sql.Node
 func (r *RecursiveCte) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
-	return &recursiveCteIter{
-		init:        r.Init,
-		rec:         r.Rec,
+	var iter sql.RowIter = &recursiveCteIter{
+		init:        r.Left(),
+		rec:         r.Right(),
 		row:         row,
 		working:     r.working,
 		temp:        make([]sql.Row, 0),
-		deduplicate: r.Deduplicate,
-	}, nil
-}
-
-// Children implements sql.Node
-func (r *RecursiveCte) Children() []sql.Node {
-	return []sql.Node{r.Init, r.Rec}
+		deduplicate: r.Distinct,
+	}
+	if r.Limit != nil && len(r.SortFields) > 0 {
```

Comment:
This is all kind of gross and duplicates transformations that should happen at analysis time (and apply to all node types wrapped by sorts / limits)

What's the rationale for this? Why not let the top-level element of the CTE be a Limit / Sort?